### PR TITLE
Add per-method flags to the v-table in class nominal type descriptors

### PR DESCRIPTION
**Pull request by rjmccall**
For now, just store a kind and whether the method was dynamic.

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please smoke test and merge.

File: include/swift/ABI/MetadataValues.h [hunk 1]
```diff
@@ -106,6 +106,47 @@ inline ClassFlags &operator|=(ClassFlags &a, ClassFlags b) {
   return a = (a | b);
 }
 
+/// Flags that go in a MethodDescriptor structure.
+class MethodDescriptorFlags {
+public:
+  typedef uint32_t int_type;
+  enum class Kind {
+    InstanceMethod,
```
Comment (slavapestov) [inline include/swift/ABI/MetadataValues.h]: I feel like static vs instance is orthogonal to method vs accessor, given that class property accessors can appear in the vtable as well.

Comment (rjmccall) [reply include/swift/ABI/MetadataValues.h]: Ah, yeah, that's true.  Maybe 'static' makes more sense as a separate bit.

Comment (rjmccall) [reply include/swift/ABI/MetadataValues.h]: I was definitely just not thinking about static properties.

Comment (rjmccall) [reply include/swift/ABI/MetadataValues.h]: Maybe it could be a field, like static vs. mutating vs. non-mutating vs. consuming?

Comment (rjmccall) [reply include/swift/ABI/MetadataValues.h]: On the other hand, maybe that belongs more in a type description.

Comment (rjmccall) [reply include/swift/ABI/MetadataValues.h]: Submitted #11712 to address this.

**Pull request merged**